CppImport/src/ClangHelpers.cpp

```diff
@@ -290,4 +293,57 @@ void ClangHelpers::attachDeclarationComments(clang::NamedDecl* namedDecl, Model:
 	}
 }
 
+void ClangHelpers::storeNodeFilename(Model::Node* node, clang::SourceLocation location)
+{
+	auto fullPath = presumedFilenameWithoutExtension(location);
+	nodeToFilenameMap_ [node] = fullPath.right(fullPath.size() - rootProjectPath_.size() - 1);
+}
+
+void ClangHelpers::storeNodeFilename(Model::Node* node, Model::Node* from)
+{
+	auto it = nodeToFilenameMap_.find(from);
+	Q_ASSERT(it != nodeToFilenameMap_.end());
+	nodeToFilenameMap_[node] = *it;
+}
+
+void ClangHelpers::dumpMergeMap(QString filename)
+{
+	QHash<QString, QString> dummy;
+	QList<CppExport::CodeUnit*> codeUnits;
+	CppExport::CppExporter::units(rootProject_, codeUnits, dummy);
+
+	QHash<QString, QString> dumpedMergeMap;
```

Comment:
Is the dumped map minimal? Shouldn't we use the `dummy` map to check if an entry we're about to insert into the dumped map can be successfully generated automatically (and therefore ignore it completely)?